src/AutomationTests/AutomationIntegrationTests.cs

```diff
@@ -77,139 +77,182 @@ public AutomationIntegrationTests()
-            ScanIntegrationCore(sync, _webViewSampleAppPath, WebViewSampleKnownErrorCount);
+            RunWithTimedExecutionWrapper(TimeSpan.FromSeconds(60), () =>
+            {
+                ScanIntegrationCore(sync, _webViewSampleAppPath, WebViewSampleKnownErrorCount);
+            });
         }
 
         [TestMethod]
-        [Timeout(30000)]
         public void ScanAsync_WindowsFormsSampler_TaskIsCancelled_ThrowsCancellationException()
         {
-            var processId = LaunchTestApp(_windowsFormsControlSamplerAppPath);
-            var builder = Config.Builder.ForProcessId(processId)
-                .WithOutputDirectory(_outputDir)
-                .WithOutputFileFormat(OutputFileFormat.A11yTest);
+            RunWithTimedExecutionWrapper(TimeSpan.FromSeconds(30), () =>
+            {
+                var processId = LaunchTestApp(_windowsFormsControlSamplerAppPath);
+                var builder = Config.Builder.ForProcessId(processId)
+                    .WithOutputDirectory(_outputDir)
+                    .WithOutputFileFormat(OutputFileFormat.A11yTest);
 
-            var config = builder.Build();
+                var config = builder.Build();
 
-            var scanner = ScannerFactory.CreateScanner(config);
+                var scanner = ScannerFactory.CreateScanner(config);
 
-            var cancellationTokenSource = new CancellationTokenSource();
-            cancellationTokenSource.Cancel();
+                var cancellationTokenSource = new CancellationTokenSource();
+                cancellationTokenSource.Cancel();
 
-            var task = scanner.ScanAsync(null, cancellationTokenSource.Token);
+                var task = scanner.ScanAsync(null, cancellationTokenSource.Token);
 
-            ValidateTaskCancelled(task);
+                ValidateTaskCancelled(task);
+            });
         }
 
         [TestMethod]
-        [Timeout(45000)]
-        public async Task ScanAsync_WildlifeManager_MultipleProcesses_RunToCompletion()
+        public void ScanAsync_WildlifeManager_MultipleProcesses_RunToCompletion()
         {
-            var instanceCount = 3;
-            var cancellationTokens = Enumerable.Range(0, instanceCount).Select(_ => new CancellationTokenSource().Token).ToList();
-            var tasks = GetAsyncScanTasks(_wildlifeManagerAppPath, cancellationTokens);
-            var results = await Task.WhenAll(tasks);
-
-            foreach (var result in results)
+            const int instanceCount = 3;
+            RunWithTimedExecutionWrapper(TimeSpan.FromSeconds(20 * instanceCount), () =>
             {
-                ValidateOutput(result.WindowScanOutputs, WildlifeManagerKnownErrorCount);
-            }
+                var cancellationTokens = Enumerable.Range(0, instanceCount).Select(_ => new CancellationTokenSource().Token).ToList();
+                var tasks = GetAsyncScanTasks(_wildlifeManagerAppPath, cancellationTokens);
+                var results = Task.WhenAll(tasks).Result;
```

Comment:
Using `.Result` here (and in the next method) allowed me to keep `TimedExecutionWrapper` simple